py/desisim/simexp.py

```diff
@@ -505,18 +512,32 @@ def simulate_spectra(wave, flux, fibermap=None, obsconditions=None,
         source_fraction[elgs,0]=1.   # ELG are disk only
         source_fraction[lrgs,1]=1.   # LRG are bulge only
         source_fraction[bgss,0]=0.52 # disk comp in BGS
-        source_fraction[bgss,1]=0.48 # bulge comp in BGS
-        
+        source_fraction[bgss,1]=0.48 # bulge comp in BGS       
 
         # source_half_light_radius[:,0] is the half light radius in arcsec for the DISK profile
         # source_half_light_radius[:,1] is the half light radius in arcsec for the BULGE profile        
         # see specsim.GalsimFiberlossCalculator.create_source routine
         source_half_light_radius=np.zeros((nspec,2))
         source_half_light_radius[elgs,0]=0.45 # ELG are disk only, arcsec
         source_half_light_radius[lrgs,1]=1.   # LRG are bulge only, arcsec
-        source_half_light_radius[bgss,0]=4.7  # disk comp in BGS, arcsec
-        source_half_light_radius[bgss,1]=1.3  # bulge comp in BGS, arcsec
-       
+
+        # 4.7 is angular size of z=0.1 disk, and 1.3 is angular size of z=0.1 bulge
+        bgs_disk_z01 = 4.7
+        bgs_bulge_z01 = 1.3
+        
+        # Input cosmology to calculate the angular diameter distance of the galaxy's redshift
+        from astropy.cosmology import Planck15 as LCDM
+        ang_diam_dist = LCDM.angular_diameter_distance
```

Comment:
Two comments: (1) I would put this import at the top of the `if` statement.  And (2) I don't feel strongly about this but I would use a much simpler concordance cosmology of H0=70, Omega0=0.3, and OmegaL=0.7, rather than the Planck+15 cosmology.  My $0.02.